[IMP] events: full rewrite of sell tickets (#8928)
odoo/documentation: content/applications/marketing/events/sell_tickets.rst

==================
Sell event tickets
==================

> Not only does Odoo *Events* provide users with the ability to create custom event tickets (and
> ticket tiers), with various price points, it also allows them to sell event tickets in two
> different ways: via standard sales orders and online through an integrated website.

Reviewer: ```suggestion
Odoo *Events* provides users with the ability to create custom event tickets (and ticket tiers), with various price points.
It *also* allows them to sell event tickets in two different ways: via standard sales orders, and online through an integrated website.
```
Optional rewording

Odoo also simplifies the ticket-purchasing process by providing plenty of payment method options.

.. tip::
   To learn more about how to create custom tickets (and ticket tiers) for events, check out the
   :doc:`create_events` documentation.

Configuration
=============

In order to sell event tickets in Odoo, some settings must first be enabled.

> First, navigate to :menuselection:`Events app --> Configuration --> Settings`, and in the
> :guilabel:`Registration` section, there are two different settings: :guilabel:`Tickets` and
> :guilabel:`Online Ticketing`.

Reviewer: ```suggestion
First, navigate to :menuselection:`Events app --> Configuration --> Settings`. In the
:guilabel:`Registration` section, there are two different settings: :guilabel:`Tickets` and
:guilabel:`Online Ticketing`.
```

The :guilabel:`Tickets` setting allows users to sell event tickets with standard sales orders.

The :guilabel:`Online Ticketing` setting allows users to sell event tickets online through their
integrated Odoo website.

> To activate one (or both) of these settings, tick the checkbox beside the desired feature(s), and

Reviewer: ```suggestion
To activate one (or both) of these settings, tick the checkbox beside the desired features, and
```
It's my understanding we shouldn't use the (s) anymore, but if I'm wrong, ignore this change

click :guilabel:`Save` to implement those settings.

.. note::
   If these options are *not* enabled, a default :guilabel:`Register` button becomes available for
   visitors to interact with and procure free registrations to the event.

.. image:: sell_tickets/events-settings-tickets.png
   :align: center
   :alt: View of the settings page for Odoo Events.

> With those settings enabled and saved, Odoo automatically creates a new *Product Type* called,

Reviewer: ```suggestion
With those settings enabled, Odoo automatically creates a new *Product Type* called
```

*Event Ticket*, which is accessible on every product form. Odoo also creates three event
registration products (with the *Product Type* set to *Event Ticket*) that can be used or modified
for event tickets.

.. important::
   When creating a new event registration product, the *Product Type* **must** be set to *Event
   Ticket* in order for it to be selected in the *Product* column in the *Tickets*  tab on an event
   form.

   .. image:: sell_tickets/events-tickets-registration-product.png
      :align: center
      :alt: View of an event form highlighting the column product under the tickets tab in Odoo.

.. note::

>    Any event with tickets sold with standard sales orders, or online, features a :guilabel:`Sales`

Reviewer: ```suggestion
   Any event with tickets sold either with standard sales orders, or online, features a :guilabel:`Sales`
```

   smart button at the top of the event form.

   .. image:: sell_tickets/events-sales-smartbutton.png
      :align: center
      :alt: View of an event's form and the sales smart button in Odoo Events.

   Clicking that smart button reveals a separate page, showcasing all the sales orders (standard
   and/or online) related to tickets that have been sold for that specific event.

Sell event tickets with sales orders
====================================

> To sell event tickets with sales orders, start by navigating to :menuselection:`Sales` app. Then,

Reviewer: ```suggestion
To sell event tickets with sales orders, start by navigating to the :menuselection:`Sales` app. Then,
```
Optional

click :guilabel:`New` to open a new quotation form.

After filling out the top portion of the form with the appropriate customer information, click
:guilabel:`Add a product` in the :guilabel:`Order Lines` tab. Then, select (or create) an event
registration product configured with its *Product Type* set to *Event Ticket* on its product form.

Once an event registration product is selected, a :guilabel:`Configure an event` pop-up window
appears.

.. image:: sell_tickets/configure-event-popup.png
   :align: center
   :alt: Standard 'Configure an event' pop-up window that appears on an event ticket sales order.

From the :guilabel:`Configure an event` pop-up window, select to which event this ticket purchase is
related to in the :guilabel:`Event` field drop-down menu. Then, in the :guilabel:`Event Ticket`

> drop-down field, select which ticket tier (if there are multiple configured for that event) the
> customer wishes to purchase.

Reviewer: ```suggestion
drop-down field, select which ticket tier the customer wishes to purchase, if there are multiple tiers configured for that event.
```

When all the desired configurations are complete, click :guilabel:`Ok`. Doing so returns the user to

> the sales order, with the event registration ticket product now present in the :guilabel:`Order
> Lines` tab, the user can proceed to confirm and close the sale, per the usual process.

Reviewer: ```suggestion
the sales order, with the event registration ticket product now present in the :guilabel:`Order
Lines` tab. The user can proceed to confirm and close the sale, per the usual process.
```

:guilabel:`Tickets` pop-up window.

.. image:: sell_tickets/tickets-popup.png
   :align: center
   :alt: The tickets pop-up window that appears on the event's website when 'Register' is clicked.

From here, visitors select which ticket tier they would like to purchase, along with a quantity,
using the numerical drop-down menu available to the right of their desired ticket. Once the desired
selections have been entered, the visitor then clicks the :guilabel:`Register` button.

Then, an :guilabel:`Attendees` pop-up window appears, containing all the questions that have been
configured in the *Questions* tab of the event form for this particular event.

.. image:: sell_tickets/attendees-popup.png
   :align: center
   :alt: The attendees pop-up window that appears on the event's website when 'Ok' is clicked.

If multiple tickets are being purchase at once, there are numbererd sections for each individual
ticket registrant, each containing the same questions. However, if any question has been configured
with the *Ask once per order* setting, that question is only asked once -- and **not** for every
attendee making the reservation in the order.

> With all necessary information (and responses) entered, the visitor can then click the :guilabel:`Go

Reviewer: ```suggestion
With all necessary information entered, the visitor can then click the :guilabel:`Go
```
I think "information" encompasses everything

The :guilabel:`Tickets` setting allows users to sell event tickets with standard sales orders.

The :guilabel:`Online Ticketing` setting allows users to sell event tickets online through their
integrated Odoo website.

To activate one (or both) of these settings, tick the checkbox beside the desired features, and
click :guilabel:`Save` to implement those settings.

.. note::
   If these options are *not* enabled, a default :guilabel:`Register` button becomes available for
   visitors to interact with and procure free registrations to the event.

.. image:: sell_tickets/events-settings-tickets.png
   :align: center
   :alt: View of the settings page for Odoo Events.

With those settings enabled, Odoo automatically creates a new *Product Type* called, *Event Ticket*,
which is accessible on every product form. Odoo also creates three event registration products (with
the *Product Type* set to *Event Ticket*) that can be used or modified for event tickets.

.. important::
   When creating a new event registration product, the *Product Type* **must** be set to *Event
   Ticket* in order for it to be selected in the *Product* column in the *Tickets*  tab on an event

>    form.

Reviewer 2: Consider adding some specificity here because we're talking about two different forms in the same admonition block (product form vs. events form > tickets tab). Think we're missing some guis, as well? 🤔

   .. image:: sell_tickets/events-tickets-registration-product.png
      :align: center
      :alt: View of an event form highlighting the column product under the tickets tab in Odoo.

> .. note::
>    Any event with tickets sold either with standard sales orders, or online, features a
>    :guilabel:`Sales` smart button at the top of the event form.

Reviewer 2: There are [examples](https://www.odoo.com/web#id=5153&cids=3&menu_id=4969&action=1986&model=event.event&view_type=form) inside Odoo where that smart button is missing despite paid tickets being claimed. I wonder why 🤔 

Besides that, I get what you were saying in the **Note**, however I do think the language is a bit off, because whether or not we're using the Website application to purchase tickets, a "standard sales order" is still generated in Odoo. Do you see what I mean?

Consider something like this, instead?
```suggestion
.. note::
   Any event with paid tickets sold will feature a :guilabel:`Sales` smart button at the top of the
   event form, where the respective sales orders attributed to those ticket sales become available.
```

>    Any event with paid tickets sold will feature a :guilabel:`Sales` smart button at the top of the
>    event form, where the respective sales orders attributed to those ticket sales become available.

Reviewer 3: Suggestion to add the icon to the smart button :)
```suggestion
   Any event with paid tickets sold, features a :icon:`fa-dollar` :guilabel:`Sales` smart button at
   the top of the event form, where the respective sales orders attributed to those ticket sales
   become available.
```

   .. image:: sell_tickets/events-sales-smartbutton.png
      :align: center
      :alt: View of an event's form and the sales smart button in Odoo Events.

>    Clicking that smart button reveals a separate page, showcasing all the sales orders (standard
>    and/or online) related to tickets that have been sold for that specific event.

Reviewer 3: Icon here too if you'd like :)
```suggestion
   Clicking the :icon:`fa-dollar` :guilabel:`Sales` smart button reveals a separate page, showcasing
   all the sales orders (standard and/or online) related to tickets that have been sold for that
   specific event.
```

> Sell event tickets with sales orders

Reviewer 2: Same distinction in the last comment.
All events tickets involve sales orders, the difference is the apps used to generate them. 

Consider editing the language to this and the next section accordingly.

```suggestion
Sell event tickets with the Sales app
```

=====================================

To sell event tickets with sales orders, start by navigating to the :menuselection:`Sales` app.
Then, click :guilabel:`New` to open a new quotation form.

After filling out the top portion of the form with the appropriate customer information, click

> :guilabel:`Add a product` in the :guilabel:`Order Lines` tab. Then, select (or create) an event
> registration product configured with its *Product Type* set to *Event Ticket* on its product form.

Reviewer 3: Specify the Product column, just to be precise. And since you stated "(or create)" I think it is safe to use :guilabels: for the field and value on the product form :)
```suggestion
:guilabel:`Add a product` in the :guilabel:`Order Lines` tab. Then, in the :guilabel:`Product`
column, select (or create) an event registration product configured with its :guilabel:`Product
Type` set to :guilabel:`Event Ticket` on its product form.
```

Once an event registration product is selected, a :guilabel:`Configure an event` pop-up window
appears.

.. image:: sell_tickets/configure-event-popup.png
   :align: center
   :alt: Standard 'Configure an event' pop-up window that appears on an event ticket sales order.

From the :guilabel:`Configure an event` pop-up window, select to which event this ticket purchase is
related to in the :guilabel:`Event` field drop-down menu. Then, in the :guilabel:`Event Ticket`

> drop-down field, select which ticket tier the customer wishes to purchase, if there are multiple

Reviewer 3: Semantics– the same field type (Event) was referred to as a drop-down menu
```suggestion
drop-down menu, select which ticket tier the customer wishes to purchase, if there are multiple
```

tiers configured for that event.

When all the desired configurations are complete, click :guilabel:`Ok`. Doing so returns the user to
the sales order, with the event registration ticket product now present in the :guilabel:`Order
Lines` tab. The user can proceed to confirm and close the sale, per the usual process.

> 

Reviewer 3: Suggestion to add a tip on how to re-open the Configure an event pop-up
```suggestion
.. tip::
   To re-open the *Configure an event* pop-up window, hover over the event registration product name
   in the :guilabel:`Order Lines` tab, and click on the :icon:`fa-pencil` :guilabel:`(pencil)` icon.

```

> Sell event tickets online

Reviewer 2: ```suggestion
Sell event tickets through the Website app
```

==========================================

When a visitor arrives on the register page of the event website, they can click the
:guilabel:`Register` button to purchase a ticket to the event.

.. note::
   If the visitor is *not* already on the register page of the event website, clicking
   :guilabel:`Register` on the event website's submenu redirects them to the proper

>    :guilabel:`Register` page. From there, they can click the :guilabel:`Register` button to begin

Reviewer 3: No :guilabel: for "register page"
```suggestion
   register page. From there, they can click the :guilabel:`Register` button to begin
```

   the ticket purchasing process.

If different ticket tiers are configured for the event, the visitor is presented with a
:guilabel:`Tickets` pop-up window.

.. image:: sell_tickets/tickets-popup.png
   :align: center
   :alt: The tickets pop-up window that appears on the event's website when 'Register' is clicked.

From here, visitors select which ticket tier they would like to purchase, along with a quantity,
using the numerical drop-down menu available to the right of their desired ticket. Once the desired
selections have been entered, the visitor then clicks the :guilabel:`Register` button.

Then, an :guilabel:`Attendees` pop-up window appears, containing all the questions that have been
configured in the *Questions* tab of the event form for this particular event.

.. image:: sell_tickets/attendees-popup.png
   :align: center
   :alt: The attendees pop-up window that appears on the event's website when 'Ok' is clicked.

> If multiple tickets are being purchase at once, there are numbererd sections for each individual

Reviewer 3: Typo
```suggestion
If multiple tickets are being purchase at once, there are numbered sections for each individual
```

ticket registrant, each containing the same questions. However, if any question has been configured
with the *Ask once per order* setting, that question is only asked once -- and **not** for every
attendee making the reservation in the order.

With all necessary information entered, the visitor can then click the :guilabel:`Go to Payment`

> button. Doing so first takes the visitor to a *Billing* confirmation page, followed by a *Payment*
> confirmation page, where they can utilze any configured payment method set up in the database to
> complete the order.

Reviewer 3: I think it is safe to use guilabels here as well since the user is proceeding through the purchasing flow. Also fix typo :)
```suggestion
button. Doing so first takes the visitor to a :guilabel:`Billing` confirmation page, followed by a
:guilabel:`Payment` confirmation page, where they can utilize any configured payment method set up
in the database to complete the order.
```
